### PR TITLE
Remove core Elasticsearch dependency

### PR DESCRIPTION
**Pull request by sjudeng**
This PR removes support for the Elasticsearch Transport client and updates to manually build request objects instead of using the ES Java API. This allows for the removal of the `org.elasicsearch:elasticsearch` dependency and the addition of compatibility shims that resolve deprecation warnings under Elasticsearch 5.x and also restores support for Elasticsearch 1.x without requiring custom builds.

Related issue is #92.

### CODE REVIEW

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/compat/ES2Compat.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private static final IndexFeatures FEATURES = coreFeatures().supportsGeoContains().build();
+
+    @Override
+    public String scriptLang() {
+        return "groovy";
+    }
+
+    @Override
+    public Map<String,Object> createKeywordMapping() {
+        return ImmutableMap.of(ES_TYPE_KEY, "string", ES_INDEX_KEY, "not_analyzed");
```
Comment (amcp) [inline janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/compat/ES2Compat.java]: externalize string constant

File: docs/versions.txt [hunk 2]
```diff
@@ -13,7 +13,7 @@ JanusGraph.
 [options="header"]
 |==========================
 | JanusGraph | Cassandra | HBase | Bigtable | Elasticsearch | Solr | TinkerPop
-| 0.1.0 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z | 0.9.z | 2.z,5.z* |  5.2.z | 3.2.z |
+| 0.1.0 | 1.2.z, 2.0.z, 2.1.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z | 0.9.z | 1.z*,2.z,5.z |  5.2.z | 3.2.z |
```
Comment (sjudeng) [inline docs/versions.txt]: No we dropped support for 1.z with previous update (#79) and this PR restores it. But unit tests require manually starting a 1.z ES instance so it'll be tough to maintain the support for new features/updates. Should we take 1.z back out of the table?

Comment (amcp) [inline docs/versions.txt]: Version compatibility is different from ease of testing, so I think 1.z can stay in this list.

**Comment by sjudeng** [comment]
Please hold off on merging this until #233 is merged.

**Comment by amcp** [comment]
@sjudeng ok to rebase and resolve conflicts here.

**Pull request merged**